packages/flutter_tools/lib/runner.dart

```diff
@@ -111,10 +111,20 @@ Future<int> run(
 
           // TODO(eliasyishak): Set the telemetry for the unified_analytics
           //  package as well, the above will be removed once we have
-          //  fully transitioned to using the new package
+          //  fully transitioned to using the new package, https://github.com/flutter/flutter/issues/128251
           await globals.analytics.setTelemetry(true);
         }
 
+        // Ensure that anyone opted out of package:unified_analytics is also
+        // opted out of legacy analytics and that analytics is not being suppressed
+        // TODO(eliasyishak): remove once GA3 sunset, https://github.com/flutter/flutter/issues/128251
+        if (!globals.analytics.telemetryEnabled &&
+            globals.flutterUsage.enabled &&
+            !globals.flutterUsage.suppressAnalytics &&
+            globals.analytics is! NoOpAnalytics) {
```

Comment:
Needing to do a type assertion here feels like a code smell (ideally this code should not need to know about any of the concrete implementation types of `Analytics`). Why do we need to explicitly exclude `NoOpAnalytics`?

Reply:
Yeah i think this isn't ideal either, i just noticed that there were some breaking devicelab tests because this logic was opting out GA3.

In the devicelab, the tests use a log file version for the GA3 and package:unified_analytics automatically gets assigned as `NoOpAnalytics`.  I can see if i can find an alternative

[Example in the device lab test where we check the log file for events](https://github.com/flutter/flutter/blob/main/dev/devicelab/bin/tasks/module_test.dart#L313-L315)

Reply:
I removed this the type check and instead use the client id to check that we don't have a NoOp instance created.

This has all the tests passing now. Would you prefer that I fix the devicelab tests instead?

Reply:
Can you fix the devicelab tests. In general, rather than adding production code specifically to satisfy test setups, we should prefer more awkward test code so that our production code is as simple as possible. Ideally, flutter_tools would know nothing about the `NoOpAnalytics` class.

Reply:
That makes sense, I removed that check from the devicelab test since this new workflow in this PR prevents events from being sent, even when using a log file